### ISSUE TITLE
Brainstorming

### ISSUE THREAD

**Issue by Derwaan** [cursemeslowly/devrantia#3]
> I open this issue for pure brainstorming purpose only. The idea behind this issue is to propose a lot of ideas and maybe discuss them in another issue. Feel free to close it if this is not appropriate :wink: 

So here are some idea that I had will reading the wiki:
- **About the Devranter**
  - Roles with specific accessible skills (front-end developer, back-end developer)
  - Another way around, gain roles by learning skills

- **About Tasks**
  - Have requirements to access a task: need to have the skill `PHP` before accessing the task `Learn Laravel`
  - Group tasks for boss-fight like
  - Collaborate with friends to beat task/boss
  - Global task for word-boss like (every player works together on this task)

- **About Tools**
  - Trading system
  - Public market 
  - Craft to build your own hardware

- **About the Dev Space**
  - Customization (show your earned awards in your house)
  - Invite a friend to visit

Probably, you have already thought about these ideas. Also, some could seem unfeasible but better keep dreaming!

**Comment by cursemeslowly** [comment]
Yes very good. I have only thought of very basic. You have mentioned very good points.
Need more of this.
